crypto/asn1/a_mbstr.c

```diff
@@ -272,12 +270,15 @@ static int out_utf8(unsigned long value, void *arg)
 static int type_str(unsigned long value, void *arg)
 {
     unsigned long types;
+    const int native = ossl_fromascii(value);
```

Comment:
As a general point, not necessarily relevant here, converting from an unsigned value that doesn't fit into the target signed type can be done in a defined fashion by a double-subtraction sort of scheme, so that you do the conversion in the [0, INT_MAX] range which is defined, and then subtract down so that the value originally in [INT_MAX+1, UINT_MAX] ends up in [INT_MIN, -1].  But of course you have to put in the right conditionals to do so at the right times, it's quite messy, you have to be careful about the details, and it's generally more trouble than people bother to go to.